Trine.Analyzer/EnumWithoutZeroAnalyzer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Trine.Analyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class EnumWithoutZeroAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = DiagnosticIds.EnumWithoutZero;
+
+        private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, "Enum value must not be zero", "Enum value must not be zero", "Category", DiagnosticSeverity.Warning, isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.EnumDeclaration);
+        }
+
+        private static void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            var semanticModel = context.SemanticModel;
+            var zeroConstant = SyntaxFactory.LiteralExpression(
+                SyntaxKind.NumericLiteralExpression,
+                SyntaxFactory.Literal(0)
+            );
+            
+            var enumSyntax = (EnumDeclarationSyntax)context.Node;
+            var hasEnumWithZeroValue = enumSyntax.Members
+                .Any(member =>
+                {
+                    if (member.EqualsValue == null) return false;
```

Comment:
Not sure and I haven't run this analyzer code but would this enum pass or fail? My suspicion is fail since there are not values at all
```csharp
public enum DummyEnum
{
    ThisMeansZero,
    GreaterThanZero,
    AnotherOne
}
```

Reply:
This validation only handles explicit values. It's a bit more complicated to calculate implicit values, and since we already have another analyser for missing enum values I didn't think it was worth implementing, ie we'll get a validation error in any way. Good enough?

Reply:
Coolio, then we have it covered :100: